src/transformState.js

```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const key in actions) {
+    if (actions[key].type === 'clear') {
+      for (const property in state) {
+        delete state[property];
+      }
+    }
+
+    if (actions[key].type === 'addProperties') {
+      Object.assign(state, actions[key].extraData);
+    }
+
+    if (actions[key].type === 'removeProperties') {
+      for (const azione in actions[key].keysToRemove) {
+        delete state[actions[key].keysToRemove[azione]];
```

Comment:
split this single-line action into multi-line actions so that it becomes more readable